Makefile

```diff
@@ -90,7 +90,7 @@ endif
 run-e2e-tests:
 	KUBECONFIG=${kubeconfig} go test -mod=$(MOD) -tags="$(platform),e2e" -covermode=atomic -buildmode=exe -v ./test/...
 	# Test if the metrics are actually being scraped
-	KUBECONFIG=${kubeconfig} go test -mod=$(MOD) -tags="$(platform),poste2e" -covermode=atomic -buildmode=exe -v ./test/...
+	KUBECONFIG=${kubeconfig} PLATFORM=${platform} go test -mod=$(MOD) -tags="$(platform),poste2e" -covermode=atomic -buildmode=exe -v ./test/...
```

Comment:
If we take platform as env variable now, I wonder if we could somehow get rid of passing it as a build tag completely :smile: On the other hand, controlling the flow using build tags reduces the amount of code which needs to be compiled :man_scientist:

Reply:
> On the other hand, controlling the flow using build tags reduces the amount of code which needs to be compiled

It don't think this is a big factor to consider when e2e tests

Reply:
How about we solve those problems in a separate PR than this PR? Because we are holding up this PR for quite a long time now?